src/continuous_agg.c

```diff
@@ -886,21 +892,36 @@ ts_continuous_agg_watermark(PG_FUNCTION_ARGS)
 	Dimension *dim;
 	Datum maxdat;
 	bool max_isnull;
-	int64 watermark;
+	Watermark *watermark = (Watermark *) fcinfo->flinfo->fn_extra;
```

Comment:
Not really obvious the lifetime of this. For example, will this field always be NULL when this function is called directly? (in tests for example).

Reply:
`fn_extra` is always NULL on first call of the function. https://www.postgresql.org/docs/current/plhandler.html